Support language-tagged strings in create step

langsamu opened the pull request:
Resolves nationalarchives/kettle-jena-plugins#7 .

Step property creation algorithm handles new case when datatype is missing and language is set.
Dialog shows new language column.
Step meta serialises new language data.
New integration test creates statement with language-tagged string literal object.

langsamu committed:
Support language-tagged strings in create step

Step property creation algorithm handles new case when datatype is missing and language is set.
Dialog shows new language column.
Step meta serialises new language data.
New integration test creates statement with language-tagged string literal object.

adamretter commented on src/test/java/uk/gov/nationalarchives/pdi/step/jena/model/LanguageTagIT.java:
```diff
@@ -0,0 +1,91 @@
+        expected.add(
+                expected.createResource("http://example.com/s"),
+                expected.createProperty("http://example.com/p"),
+                expected.createLiteral("o","en"));
```
I would like to see further checks here, e.g. literals with and without languages, and also at least two different languages please.

adamretter commented on src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStepMeta.java:
```diff
@@ -107,6 +108,7 @@
         String fieldName;
         QName rdfPropertyName;
         QName rdfType;
+        String language; // TODO(SL): Should this be java.util.Locale?
```
Is the language optional? If not what is the default?

If it is optional we should probably mark it as `@Nullable` and with the serialization/deserialization we need to handle that.

langsamu replied:
> Is the language optional? If not what is the default?

Language is optional.

> If it is optional we should probably mark it as `@Nullable`

I tried annotating with `@Nullable`, Kettle doesn't seem to care either way.
Should I mark it?
How about others fields on this class?
How should I annotate required with (if at all)?
In my understanding, field name and property name are required, type and language are optional.

> with the serialization/deserialization we need to handle that.

Serialisation/deserialisation handles both cases:
https://github.com/nationalarchives/kettle-jena-plugins/blob/d95bb22a2ec7f55f4c4934df632fe9d694fc65e2/src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java#L240-L246

adamretter replied:
It's in general good form to annotate potentially null fields, parameters, and return values with `javax.annotation.Nullable` i.e. `@Nullable`.

Two quick good reasons:

1. It serves as a form of code documentation. The caller of a function or accessor to a field knows what to expect.
2. It can be used in conjunction with Static code checkers like FindBugs or PMD to improve code quality through automated review.

adamretter reviewed (commented):
Great start. Just a few suggestions

langsamu committed:
Increase coverage of language-tagged strings

Type and untyped with multiple languages.
Also add missing serialiser plugin injection.

adamretter commented:
@langsamu Can you also rebase this please now that your other PR was merged?

langsamu committed:
Mark optional fields as such

langsamu committed:
Merge branch 'master' of https://github.com/nationalarchives/kettle-jena-plugins into lang-strings
